www/class/centreonTraps.class.php

```diff
@@ -187,19 +254,42 @@ public function trapNameExists($trapName)
      * @param $traps
      * @param $nbrDup
      */
-    public function duplicate($traps = array(), $nbrDup = array())
+    public function duplicate($traps = [], $nbrDup = [])
     {
+        $querySelectTrap = "SELECT * FROM traps WHERE traps_id = :trapsId LIMIT 1";
+        $queryInsertTrapServiceRelation = "
+            INSERT INTO traps_service_relation (traps_id, service_id)
+            (SELECT :maxTrapsId, service_id
+                FROM traps_service_relation
+                WHERE traps_id = :trapsId)";
+        $queryInsertPreexec = "
+            INSERT INTO traps_preexec (trap_id, tpe_string, tpe_order)
+            (SELECT :maxTrapsId, tpe_string, tpe_order
+                FROM traps_preexec
+                WHERE trap_id = :trapsId)";
+        $querySelectMatching = "SELECT * FROM traps_matching_properties WHERE trap_id = :trapsId";
+        $queryInsertMatching = "
+            INSERT INTO traps_matching_properties
+            (`trap_id`,`tmo_order`,`tmo_regexp`,`tmo_string`,`tmo_status`,`severity_id`)
+            VALUES (:trap_id, :tmo_order, :tmo_regexp, :tmo_string, :tmo_status, :severity_id)";
+
+        $stmtSelectTrap = $this->db->prepare($querySelectTrap);
+        $stmtInsertTrapServiceRelation = $this->db->prepare($queryInsertTrapServiceRelation);
+        $stmtInsertPreexec = $this->db->prepare($queryInsertPreexec);
+        $stmtSelectMatching = $this->db->prepare($querySelectMatching);
+        $stmtInsertMatching = $this->db->prepare($queryInsertMatching);
         foreach (array_keys($traps) as $trapsId) {
             if (is_int($trapsId)) {
-                $res = $this->db->query(
-                    "SELECT * FROM traps WHERE traps_id = $trapsId LIMIT 1"
-                );
-                $trapConfigurations = $res->fetchRow();
+
+                $stmtSelectTrap->bindValue(':trapsId', $trapsId, \PDO::PARAM_INT);
```

Comment:
sure it's bindvalue and not bindparam ?

Reply:
it works properly like this, why $trapsId could be passed as reference with bindParam ?